### DISCUSSION TITLE
Pico 4 Status

### DISCUSSION THREAD

**Discussion by phr00t** [korejan/ALVR#48] [General]
I'm looking forward to getting my Pico 4 in about a week. I'm curious, what is the status of this working with the Pico 4? I see the latest release is back in August, which I presume doesn't work with the Pico at all. I'd like to get right into at least testing this, and I'll likely be able to help provide feedback and hopefully more.

**Comment by korejan** [comment]
Hey, latest stable release works on Pico 4, this was the version (alxr-client-pico.apk) that was used to confirm this from the first day of release (in China), you can also find a youtube & bilibili video of it running on P4.

However if you haven't seen it already there are nightly builds [here](https://github.com/korejan/ALXR-nightly/releases) which contains (important) bug fixes, optimizations and adds missing features (has support for "Foveated encoding" option).

The current status on Pico 4 is everything seems to be working as far as I've heard but the hand-tracking wasn't which i assume is/was a bug in Pico's OpenXR runtime as this code has been tested on Q1/2, Focus 3, Leap Motion and Monado's hand-tracking module. There's a few  pico specific workarounds that I had to do specifically  for PN3(L) that may not be neccessary for P4, one of which may have the affect of a very slight lag feeling to controllers (normally future times are used to query the controller poses but this hasn't worked on PN3(L)).

I'm currently working on improving the latency in general for all (or as many as possible) android flavours/devices, after this plus small feature additions there will be a new stable release.

**Reply by phr00t** [reply]
Thank you so much for the reply!

I'm not too worried about hand tracking...

I'm particularly interested in USB-C wired streaming via ALXR, as I've got a laptop that would be very inconvenient to wire a router to. I see some instructions for setting up wired -- wonder if this has been tested with the Pico 4: https://github.com/alvr-org/ALVR/wiki/Use-ALVR-through-a-USB-connection

Looking forward to latency improvements, as that is my biggest worry. Have you done any comparisons with Virtual Desktop?

**Reply by korejan** [reply]
Ah I almost forgot about this, someone on the Pico discord server tried to do this with both P4 and PN3L, unfortunately they said it didn't work with P4, this might have been related to what Pico said about only the pro model having Dev support but that is just a guess.

I don't own VD, but I heard it's running pretty good on P4 and VD has support for sliced encoding while alvr currently doesn't then again alvr has foveated encoding and that can make a big difference as someone tried this on a laptop GPU (rtx 3070m) with alxr. I don't know if VD has support for foveated encoding.

There are things that could be improved on the ALVR server-side

**Comment by eMPee584** [comment]
Interested in getting one of these, have there been noticable changes in the meantime?
